### PR TITLE
Sign in form

### PR DESCRIPTION
**Pull request by nabil-dbz**
This pull request adds the sign in form to the web page. The user can get logged in by entering their email and password!

### CODE REVIEW

**Review by zhoua0805 — APPROVED** [review]

File: planet/src/main/webapp/js/authentication.js [hunk 1]
```diff
@@ -22,18 +42,35 @@ function resetForm() {
...
+function signIn() {
+    const password = document.getElementById('password-in').value;
+    const email = document.getElementById('email-in').value;
+    resetForm('input-sign-in');
+    firebase.auth().signInWithEmailAndPassword(email, password).catch(function(error) {
         // Handle Errors here.
-        var errorCode = error.code;
-        var errorMessage = error.message;
+        console.log(error);
         // ...
```
Comment (emilyzhaoo) [inline planet/src/main/webapp/js/authentication.js]: nit: Remove unused comment

**Review by emilyzhaoo — APPROVED** [review]

File: planet/src/main/webapp/js/authentication.js [hunk 2]
```diff
@@ -22,18 +42,35 @@ function resetForm() {
...
-    firebase.auth().createUserWithEmailAndPassword(email, password).catch(function(error) {
+    firebase.auth().createUserWithEmailAndPassword(email, password)
+    .then(function() {
+        return firebase.auth().currentUser.updateProfile({
+            displayName: displayName,
+        })
+    }).catch(function(error) {
+        // Handle Errors here.
+        console.log(error);
+        // ...
```
Comment (EkkiD) [inline planet/src/main/webapp/js/authentication.js]: For errors that require the user to take action, we should tell the user about them. An alert is probably good enough for the prototype.

https://firebase.google.com/docs/reference/js/firebase.auth.Auth#createuserwithemailandpassword
https://firebase.google.com/docs/reference/js/firebase.auth.Auth#signinwithemailandpassword

**Review by EkkiD — CHANGES_REQUESTED** [review]

File: planet/src/main/webapp/js/authentication.js (hunk 2 above)
Comment (nabil-dbz) [reply planet/src/main/webapp/js/authentication.js]: Okay! I'll add that right away

File: planet/src/main/webapp/js/authentication.js [hunk 3]
```diff
@@ -22,18 +42,35 @@ function resetForm() {
...
-    firebase.auth().createUserWithEmailAndPassword(email, password).catch(function(error) {
+    firebase.auth().createUserWithEmailAndPassword(email, password)
+    .then(function() {
+        return firebase.auth().currentUser.updateProfile({
+            displayName: displayName,
+        })
+    }).catch(function(error) {
+        // Handle Errors here.
+        console.log(error);
+        alert('Create account with Email and Password failed!');
```
Comment (EkkiD) [inline planet/src/main/webapp/js/authentication.js]: different errors should be handled differently. In particular, if there is an action the user can take to fix the error, we should display a message that tells them what to do.

```suggestion
    }).catch(function(error) {
        console.log(error);
       if (error.code === 'auth/email-already-in-use') {
          alert('There is already an account with this email address. Try signing in instead.');
       } else if (error.code === 'auth/invalid-email') {
         alert('Invalid email address. Double check the spelling.');
       } else if (....) {
        ....
       } else {
            // No specific action the user can take. Show a generic message.
           alert('Create account with Email and Password failed!');
       }
```

**Review by EkkiD — CHANGES_REQUESTED** [review]

File: planet/src/main/webapp/js/authentication.js (hunk 3 above)
Comment (EkkiD) [reply planet/src/main/webapp/js/authentication.js]: See https://firebase.google.com/docs/reference/js/firebase.auth.Auth#error-codes_3 and  https://firebase.google.com/docs/reference/js/firebase.auth.Auth#error-codes_12  for the possible error codes

Comment (nabil-dbz) [reply planet/src/main/webapp/js/authentication.js]: Got it! Fixed

**Review by EkkiD — COMMENTED** [review]
LGTM

**Review by EkkiD — APPROVED** [review]
LGTM

**Pull request merged**